Fix 323

Dieterbe opened the pull request.

woodsaj commented on dataprocessor.go:
```diff
@@ -310,15 +310,24 @@ func aggMetricKey(key, archive string, aggSpan uint32) string {
+	// so that if we have quantize later (in fix()) we have a valid first point.
+	// (at this point we can't tell whether the raw series will be quantized or not)
+	if consolidator == consolidation.None {
+		fromUnix = fromUnix - interval + 1
```
This doesnt seem right to me.

assume my interval is 10, and my request is from 15 to 35.  Then i would expect quantized values at 20 and 30.
With this change, you are now requesting between 6 and 35.
If the raw values are offset by between 1 and 6, then things will work, as the adjusted from time will only match points between 11 and 16 inclusive, which all get quantized to 20
but if the offset is 7,8,9 or 0 the the request will match points at 7,8,9,10 which get quantized to 10

I think what you need to do is adjust both fromUnix and ToUnix to be on the appropriate quantized boundaries.
eg. the fromUnix=10, toUnix=30 so that any points that match are quantized to either 20 or 30

Raw | quantized | match Range
------ | ------------- | -----------------
8 | 10 | no
9 | 10 | no
10 | 10 | no
11 | 20 | yes
12 | 20 | yes
. | . | .
19 | 20 | yes
20 | 20 | yes
21 | 30 | yes
. | . | .
29 | 30 | yes
30 | 30 | yes
31 | 40 | no
32 | 40 | no

Dieterbe commented:
@woodsaj I implemented the improved solution, added more complete unit tests and also made getSeries itself call fix() which removes any confusion about getSeries() output requiring an additional fix() step.
